### ISSUE TITLE
Add language filter to new user setup wizard

### ISSUE THREAD

**Issue by Cashfl0w** [tootsuite/mastodon#6867]
I believe adding the **Filtered languages** from the preferences menu to the on-boarding of new users process whilst they are setting up their account would enhance their experience considerably.

Please could this be considered for a future update.

**Comment by Cassolotl** [comment]
I agree with this, I feel like something needs to change because the language filter on the settings page is pretty unintuitive. I just had a conversation with a newbie about this:

![screen shot 2018-03-22 at 14 39 05](https://user-images.githubusercontent.com/6607512/37777188-e13041ca-2dde-11e8-8106-2a6d5b61dd87.png)

It's unintuitive; they didn't realise that you have to check boxes for the languages you *don't* want to see, so they thought it was buggy.

I understand that at the time of being introduced there was some backlash by people whose native languages are not dominant languages. (They were concerned about [linguistic imperialism](https://en.wikipedia.org/wiki/Linguistic_imperialism).) I'm wondering if this was the reason why we have to check boxes to filter languages *out* and they're all deselected by default?

See also: *"Select all/none" on filtered languages in settings #5627*

**Comment by Mbodin** [comment]
The Amikumu app just got updated, and I think that we can inspire from their new interface.

The idea is as follows: currently, it is relatively difficult to filter out language temporarily (we have to get into the settings, and it involves several clicks). Sometimes, we just want to check toots in a language we are usually not fluent, but that we understand. Typically, if your native language is not English, French, or Japanese (I don’t have any stats, but I have the feeling that these languages are pretty dominant in Mastodon), you may see a lot of such toots in your timeline unless you filter them out. But even if you filtered them out, you may want to check them temporarily from time to time.

The new Amikumu interface includes a dropdown menu in the top of the timeline, with a language to be selected. We probably don’t want exactly the same interface—as most people in Mastodon speak more than one language—, but I think that the idea of a quick filter available in the top of a timeline panel would be a good idea (with a default option “no filter”). The advantage of such a filter is that it is easy to understand. Here is a screenschot:
<img src = "https://user-images.githubusercontent.com/1829786/37780683-ffe65ea8-2dcd-11e8-82e0-ac4f4dacee58.png" width = "50%"/>

After that, there will be the issue of the textless/emoji-only toots (often accompanied by an image or a link): maybe such a quick filter should treat them as a special case. I guess that the only way to be sure is to test such an interface. What do you think?

Maybe I should have opened a new issue for this, but it seemed related enough for me.

**Comment by Cashfl0w** [comment]
I seems there's two requests here after @Mbodin introduced another useful option.

1. Add a language setup screen to the on-boarding wizard for new users
2. Enable a temporary language filter for toots (which overrides the account settings (set above), that is reset at next login?)

It feels like from a workflow point of view that these two requests should be split into two individual issues and then linked as they're both about languages but I'll leave that for someone else who actually changes the code or maintains the issues to decide.
